StRoot/StFttClusterMaker/StFttClusterMaker.cxx

```diff
@@ -0,0 +1,429 @@
+/***************************************************************************
+ *
+ * $Id: StFttClusterMaker.cxx,v 1.4 2019/03/08 18:45:40 fseck Exp $
+ *
+ * Author: Florian Seck, April 2018
+ ***************************************************************************
+ *
+ * Description: StFttClusterMaker - class to fill the StEvent from DAQ reader:
+ * unpack raw data & save StETofHeader & StETofDigis in StETofCollection 
+ *
+ ***************************************************************************/
+#include <vector>
+#include <set>
+#include <map>
+#include <array>
+#include <algorithm>    // std::is_sorted
+
+
+#include "StEvent.h"
+#include "StEnumerations.h"
+
+#include "StFttClusterMaker.h"
+
+
+#include "StEvent/StFttRawHit.h"
+#include "StEvent/StFttCluster.h"
+#include "StEvent/StEvent.h"
+#include "StEvent/StFttCollection.h"
+
+#include "StFttDbMaker/StFttDb.h"
+
+
+//_____________________________________________________________
+StFttClusterMaker::StFttClusterMaker( const char* name )
+: StMaker( name ),
+  mEvent( 0 ),          /// pointer to StEvent
+  mRunYear( 0 ),        /// year in which the data was taken (switch at 1st Oct)
+  mDebug( false ),       /// print out of all full messages for debugging
+  mFttDb( nullptr )
+{
+    LOG_DEBUG << "StFttClusterMaker::ctor"  << endm;
+}
+
+//_____________________________________________________________
+StFttClusterMaker::~StFttClusterMaker()
+{  /* no op */
+
+}
+
+//_____________________________________________________________
+Int_t
+StFttClusterMaker::Init()
+{
+    LOG_INFO << "StFttClusterMaker::Init" << endm;
+
+    return kStOk;
+}
+
+//_____________________________________________________________
+Int_t
+StFttClusterMaker::InitRun( Int_t runnumber )
+{ 
+    mRunYear = ( runnumber + 727000 ) / 1000000 + 1999;
+
+    LOG_INFO << "runnumber: " << runnumber << "  --> year: " << mRunYear << endm;
+
+    return kStOk;
+}
+
+//_____________________________________________________________
+Int_t
+StFttClusterMaker::FinishRun( Int_t runnumber )
+{ 
+    return kStOk;
+}
+
+//_____________________________________________________________
+Int_t
+StFttClusterMaker::Finish()
+{ 
+    return kStOk;
+}
+
+
+//_____________________________________________________________
+Int_t
+StFttClusterMaker::Make()
+{ 
+    LOG_INFO << "StFttClusterMaker::Make()" << endm;
+
+
+    mEvent = (StEvent*)GetInputDS("StEvent");
+    if(mEvent) {
+        LOG_DEBUG<<"Found StEvent"<<endm;
+    } else {
+        return kStOk;
```

Comment:
"return kStWarn" would be a bit more appropriate.  Also, it makes a bit more sense to issue a DEBUG message on the exceptional case (StEvent missing) rather than the case where everything is working as expected.

Reply:
Addressed: 3686a6eea3b300c95180bdcee0ad698b5763a88c